machine_learning/neural_network.cpp

```diff
@@ -390,6 +390,11 @@ namespace machine_learning {
                                      const int &slip_lines = 1) {
                     std::ifstream in_file; // Ifstream to read file
                     in_file.open(file_name.c_str(), std::ios::in); // Open file
+                    // If there is any problem in opening file
+                    if(!in_file.is_open()) {
+                        std::cerr << "ERROR: Unable to open file: "<< file_name << std::endl;
```

Comment:
```suggestion
                        std::cerr << __func__ << ": ERROR: Unable to open file: "<< file_name << std::endl;
```
this will also print which function printed that error message